File: Backend I/Semana 7/Viajes/src/Persona.java
public class Persona {
    private String nombre;
    private String genero;

    // Constructor
    public Persona(String nombre, String genero) {
        this.nombre = nombre;
        this.genero = genero;
    }

    // Métodos getter
    public String getNombre() {
        return nombre;
    }

    public String getGenero() {
        return genero;
    }

    // Métodos setter
    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public void setGenero(String genero) {
        this.genero = genero;
    }

    @Override
    public String toString() {
        return "Nombre: " + nombre + ", Género: " + genero;
    }
}

File: Frontend I/Semana 11/js/persona.js
class Persona {
	constructor(idPersona, apellido, nombre, edad, genero) {
		this.idPersona = idPersona;
		this.apellido = apellido;
		this.nombre = nombre;
		this.edad = edad;
		this.genero = genero;
	}

	imprimirDatos() {
		console.log("Datos de la persona ");
		console.log(`Nombre -----> ${this.nombre}`);
		console.log(`Apellido -----> ${this.apellido}`);
		console.log(`Edad -----> ${this.edad}`);
		console.log(`Género -----> ${this.genero}`);
	}

	actualizarDatos(nuevosDatos) {
		if (nuevosDatos.apellido) this.apellido = nuevosDatos.apellido;
		if (nuevosDatos.nombre) this.nombre = nuevosDatos.nombre;
		if (nuevosDatos.edad) this.edad = nuevosDatos.edad;
		if (nuevosDatos.genero) this.genero = nuevosDatos.genero;
	}
}

export default Persona;
